addons/l10n_in/models/account_invoice.py

```diff
@@ -97,3 +97,40 @@ def _l10n_in_get_warehouse_address(self):
         # TO OVERRIDE
         self.ensure_one()
         return False
+
+    def _l10n_in_get_hsn_summary(self):
+        for move in self:
+            hsn_summary = {}
+
+            def grouping_key_generator(base_line, tax_values):
+                hsn = base_line['product'].l10n_in_hsn_code
```

Comment:
What if product is not defined? (It will be False?)

Reply:
instead of False we set empty string so
```suggestion
                hsn = base_line['product'].l10n_in_hsn_code or ""
```